r1cs/gadgets/std/src/groups/nonnative/nonnative_group_gadget.rs

```diff
@@ -0,0 +1,1015 @@
+use algebra::{AffineCurve, BitIterator, Field, PrimeField, ProjectiveCurve, SWModelParameters, SquareRootField, curves::short_weierstrass_jacobian::{GroupAffine as SWAffine, GroupProjective as SWProjective}};
+
+use r1cs_core::{ConstraintSystem, SynthesisError};
+
+use crate::{Assignment, ToBitsGadget, ToBytesGadget, alloc::{AllocGadget, ConstantGadget}, boolean::Boolean, fields::{FieldGadget, nonnative::nonnative_field_gadget::NonNativeFieldGadget}, prelude::EqGadget, select::{CondSelectGadget, TwoBitLookupGadget}, uint8::UInt8};
+use std::{borrow::Borrow, marker::PhantomData};
+
+use super::NonNativeGroupGadget;
+
+#[derive(Derivative)]
+#[derivative(Debug, Clone)]
+pub struct GroupAffineNonNativeGadget<
+    P: SWModelParameters<BaseField = SimulationF>,
+    ConstraintF: PrimeField,
+    SimulationF: PrimeField + SquareRootField,
+    F: FieldGadget<SimulationF, ConstraintF>,
```

Comment:
Mmm I don't think you need this F: FieldGadget template at all

Reply:
Removed unused template